databases/tests/test_transactions.py

```diff
@@ -212,3 +212,61 @@ async def test_transaction_already_closed(client: Prisma) -> None:
         await transaction.user.delete_many()
 
     assert exc.match('Transaction already closed')
+
+
+@pytest.mark.asyncio
+@pytest.mark.parametrize(
+    ('input_level', 'expected_level'),
+    [
+        pytest.param(
+            prisma.TransactionIsolationLevel.READ_UNCOMMITTED,
+            'READ_UNCOMMITTED',
+            id='read uncommitted',
+            marks=pytest.mark.skipif(CURRENT_DATABASE in ['cockroachdb', 'sqlite'], reason='Not available'),
+        ),
+        pytest.param(
+            prisma.TransactionIsolationLevel.READ_COMMITTED,
+            'READ_COMMITTED',
+            id='read committed',
+            marks=pytest.mark.skipif(CURRENT_DATABASE in ['cockroachdb', 'sqlite'], reason='Not available'),
+        ),
+        pytest.param(
+            prisma.TransactionIsolationLevel.REPEATABLE_READ,
+            'REPEATABLE_READ',
+            id='repeatable read',
+            marks=pytest.mark.skipif(CURRENT_DATABASE in ['cockroachdb', 'sqlite'], reason='Not available'),
+        ),
+        pytest.param(
+            prisma.TransactionIsolationLevel.SNAPSHOT,
+            'SNAPSHOT',
+            id='snapshot',
+            marks=pytest.mark.skipif(True, reason='Available for SQL Server only'),
+        ),
+        pytest.param(
+            prisma.TransactionIsolationLevel.SERIALIZABLE,
+            'SERIALIZABLE',
+            id='serializable',
+            marks=pytest.mark.skipif(
+                CURRENT_DATABASE == 'sqlite', reason='PRAGMA has only effect in shared-cache mode'
+            ),
+        ),
+    ],
+)
+# TODO: remove after issue will be resolved
+@pytest.mark.skipif(CURRENT_DATABASE in ['mysql', 'mariadb'], reason='https://github.com/prisma/prisma/issues/22890')
+async def test_isolation_level(
+    client: Prisma, raw_queries: RawQueries, input_level: prisma.TransactionIsolationLevel, expected_level: str
+) -> None:
+    """A transaction isolation level is set correctly"""
+    async with client.tx(isolation_level=input_level) as tx:
+        results = await tx.query_raw(raw_queries.select_tx_isolation)
+
+    assert len(results) == 1
+
+    row = results[0]
+    assert any(row)
+
+    level = next(iter(row.values()))
+    # The result can depends on the database, so we do upper() and replace()
+    level = str(level).upper().replace(' ', '_').replace('-', '_')
+    assert level == expected_level
```

Comment:
@RobertCraigie I updated the tests and found a bug in Prisma 😅 

P.S. Look please at [mypy (3.11)](https://github.com/RobertCraigie/prisma-client-py/actions/runs/7723046190/job/21052347877?pr=878) step why it doesn't pass.

Reply:
Oh wow, thanks for reporting that!

The mypy failure is unrelated to your PR, I'll look into fixing it.

Reply:
If you rebase against main, the mypy job should pass now.

Reply:
I'll update docs, then wait for a response from the Prisma team, and if all is ok, the PR will be ready to final review 🎉

Reply:
> I'll update docs, then wait for a response from the Prisma team, and if bug will be confirmed, then PR will be ready to final review 🎉

Comment: https://github.com/prisma/prisma/issues/22890#issuecomment-1923606830

Reply:
of course mysql doesn't even have a way to query that 🤦 

I think it would be fine to just skip the tests for mysql if you haven't already.

Reply:
Sorry, I didn't know that...

It marked with [skipif](https://github.com/RobertCraigie/prisma-client-py/blob/577d1f112a83f909de0ec351e00675c9fd6fe0e1/databases/tests/test_transactions.py#L255-L264).